### ISSUE TITLE
Are there any reverse-proxy features like HttpServletRequest.getRequestDispatcher().forward?

### ISSUE THREAD

**Issue by anqurvanillapy** [apache/servicecomb-java-chassis#1836]
# TL;DR

* Is `HttpServletRequest.getRequestDispatcher().forward` supported?
* If not, why?
* Any alternative APIs to `forward` for reverse-proxy?

# Context

We're planning to create a new gateway service that is much like EdgeService and Nginx, basically doing reverse-proxy for internal services, but with heavy authentication and authorization due to the exposure to public network.

We have an existing project to take some reference but it's based on plain old Spring MVC, hence its lacking of service registry makes us consider the use of ServiceComb.  But eventually the `forward` API throws a not-supported exception from ServiceComb.

All solutions we found currently:

* Nginx: It's hard to integrate as a service registered to ServiceComb center
* Zuul: May be the best solution in the future, but now we're not available for doing research
* Controller + a handwritten Vert.x HTTP client to forward requests: May have poor robustness
* Spring MVC's `forward`: Seems not supported by ServiceComb

**Comment by liubao68** [comment]
For sultion 3 you listed, you can create a controller in edge service, and this controller forward to traget service. 

Or maybe you can add an custom `AbstractEdgeDispatcher` to do this. Edge service provide many `AbstractEdgeDispatcher` implementations and all disptach rule are based on these implementation.

**Comment by anqurvanillapy** [comment]
@liubao68 What if the **target service** is not based on ServiceComb, that is, without any service registry and required to be invoked via `HOST:IP`?  It's an important reason why we chose to use `forward`/HTTP client to deliver the requests.

**Comment by liubao68** [comment]
You can take 3rd party services same as servicecomb service, please refer to [this](http://localhost:8000/registry/multi-registries/#_2)

This way need configure swaggers in edge service, and it's helpful when 3rd party services count is not that huge. 

This feature will be available in 2.1.0, however you can use it in versions before in  quite a different way, see links in that doc too.

**Comment by anqurvanillapy** [comment]
Just curious, what if the target service is another edge service?  Or worse, it is simply an Nginx instance that is impossible to have a concrete set of contracts?  We're now getting to have Spring Boot + Zuul as the final solution in the future, actually.
